pkg/storage/utils/eosfs/eosfs.go

```diff
@@ -960,9 +987,11 @@ func (fs *eosfs) CreateReference(ctx context.Context, p string, targetURI *url.U
 	if err != nil {
 		return nil
 	}
-	if err := fs.createUserDir(ctx, u, tmp, false); err != nil {
-		err = errors.Wrapf(err, "eos: error creating temporary ref file")
-		return err
+
+	err = fs.c.CreateDir(ctx, uid, gid, tmp)
```

Comment:
We're using the root uid and gid here. The user won't be able to delete references. We'll need to use user credentials here but since we're setting system xattrs, we'll need root creds for that

Reply:
You're right, the reference needs to be deleted, but the user should not own any of these areas, all these folders should be own by root